### PR TITLE
chore: added mongodb and modified package and classe name TISNEW-3946

### PR DESCRIPTION
**Pull request by jayanta2018**
Added mongo db config
Changed the server port no
Changed the classes and packages

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,5 +37,5 @@ This project uses [Semantic Versioning](semver.org).
 ## License
 This project is license under [The MIT License (MIT)](LICENSE).
 
-[TemplateApplication]: src/main/java/uk/nhs/hee/tis/template/TemplateApplication.java
-[TemplateApplicationTest]: src/test/java/uk/nhs/hee/tis/template/TemplateApplicationTest.java
+[TemplateApplication]: src/main/java/uk/nhs/hee/trainee/forms/TraineeFormsApplication.java
```
Comment (Judge40) [inline README.md]: This, along with the other template instructions can be removed as it isn't relevant outside of the template.

File: build.gradle [hunk 2]
```diff
@@ -10,7 +10,7 @@ plugins {
 }
 
 // TODO: Update group to end with 'admin' or 'trainee'.
-group = 'uk.nhs.hee.tis'
+group = 'uk.nhs.hee.trainee.forms'
```
Comment (Judge40) [inline build.gradle]: package should be `uk.nhs.hee.tis.trainee.forms` and the TODO needs to be removed.

File: src/main/java/uk/nhs/hee/trainee/forms/TraineeFormsApplication.java [hunk 3]
```diff
@@ -19,16 +19,16 @@
...
  */
 
-package uk.nhs.hee.tis.template;
+package uk.nhs.hee.trainee.forms;
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
 //TODO: Update package and class name.
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/trainee/forms/TraineeFormsApplication.java]: Remove TODO

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -1,3 +1,9 @@
 server:
   # TODO: Update server port to a unique value.
-  port: 8080
+  port: 8207
+spring:
+  data:
+    mongodb:
+      host: localhost
```
Comment (Judge40) [inline src/main/resources/application.yml]: Need to include env vars for these, see https://github.com/Health-Education-England/TIS-TRAINEE-DETAILS/blob/chore/buildPipeline/src/main/resources/application.properties

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -1,3 +1,9 @@
 server:
   # TODO: Update server port to a unique value.
```
Comment (Judge40) [inline src/main/resources/application.yml]: Remove TODO

File: src/test/java/uk/nhs/hee/trainee/forms/TraineeFormsApplicationTest.java [hunk 6]
```diff
@@ -19,14 +19,14 @@
...
  */
 
-package uk.nhs.hee.tis.template;
+package uk.nhs.hee.trainee.forms;
 
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
 
 @SpringBootTest
 //TODO: Update package and class name.
```
Comment (Judge40) [inline src/test/java/uk/nhs/hee/trainee/forms/TraineeFormsApplicationTest.java]: Remove TODO

File: build.gradle [hunk 7]
```diff
@@ -62,7 +64,7 @@ sonarqube {
     property "sonar.login", System.getenv("SONAR_TOKEN")
     property "sonar.organization", "health-education-england"
     // TODO: Update sonar.projectKey to real value.
```
Comment (Judge40) [inline build.gradle]: Remove TODO

File: build.gradle (hunk 2 above)
Comment (Judge40) [reply build.gradle]: Make sure to update across the Java classes

Comment (Judge40) [reply build.gradle]: Made a mistake above and have now edited it.

File: README.md (hunk 1 above)
Comment (jayanta2018) [reply README.md]: ok removed

File: build.gradle (hunk 2 above)
Comment (jayanta2018) [reply build.gradle]: Changed accordingly

File: build.gradle (hunk 7 above)
Comment (jayanta2018) [reply build.gradle]: removed

File: src/main/java/uk/nhs/hee/trainee/forms/TraineeFormsApplication.java (hunk 3 above)
Comment (jayanta2018) [reply src/main/java/uk/nhs/hee/trainee/forms/TraineeFormsApplication.java]: removed

File: src/main/resources/application.yml (hunk 4 above)
Comment (jayanta2018) [reply src/main/resources/application.yml]: done

File: src/main/resources/application.yml (hunk 5 above)
Comment (jayanta2018) [reply src/main/resources/application.yml]: removed

File: src/test/java/uk/nhs/hee/trainee/forms/TraineeFormsApplicationTest.java (hunk 6 above)
Comment (jayanta2018) [reply src/test/java/uk/nhs/hee/trainee/forms/TraineeFormsApplicationTest.java]: removed

File: build.gradle [hunk 8]
```diff
@@ -9,8 +9,7 @@ plugins {
   id "org.sonarqube" version "2.8"
 }
 
-// TODO: Update group to end with 'admin' or 'trainee'.
-group = 'uk.nhs.hee.tis'
+group = 'uk.nhs.hee.tis.trainee.forms'
```
Comment (Judge40) [inline build.gradle]: Remove `.forms` from the group, this is different from your standard base package and should instead reflect the same sort of grouping as in Maven: https://github.com/Health-Education-England/TIS-TCS/blob/master/pom.xml#L6

Comment (jayanta2018) [reply build.gradle]: So it is a groupId of maven. Nice to know. Thanks

Comment (jayanta2018) [reply build.gradle]: removed

File: src/main/resources/application.yml [hunk 9]
```diff
@@ -1,3 +1,10 @@
 server:
-  # TODO: Update server port to a unique value.
-  port: 8080
+  port: 8207
+spring:
+  data:
+    mongodb:
+      host: ${DB_HOST:localhost}
+      port: ${DB_PORT:27017}
+      database: ${DB_NAME:trainee}
```
Comment (hihilary) [inline src/main/resources/application.yml]: the DB_NAME needs to be changed to the new db name later on.

**Pull request merged**